api/src/server.ts

```diff
@@ -52,6 +59,21 @@ const start = async () => {
       credentials: true,
     },
   });
+
+  io.on('connection', async socket => {
+    socket.on('meeting:join', meetingId => {
+      const { principalId } = socket.request.session;
+      if (participantExists(meetingId, principalId)) {
```

Comment:
`participantExists` is an async function, so this only evaluates as a truthy value because it's a Promise object, which will always be truthy.

The function on line `64` must be marked `async`, and this call to `participantExists` must be marked `await`.

Reply:
Ok will fix it now